### PR TITLE
CI cuda

### PR DESCRIPTION
**Pull request by Delaunay**

### CODE REVIEW

**Comment by Delaunay** [comment]
Failure report example

```
Error Report
------------

efficientnet_b4.D0
^^^^^^^^^^^^^^^^^^
    torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 46.00 MiB (GPU 0; 11.92 GiB total capacity; 11.24 GiB already allocated; 5.69 MiB free; 11.46 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

convnext_large.D0
^^^^^^^^^^^^^^^^^
    torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 74.00 MiB (GPU 0; 11.92 GiB total capacity; 11.32 GiB already allocated; 50.69 MiB free; 11.42 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

regnet_y_128gf.D0
^^^^^^^^^^^^^^^^^
    torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 36.00 MiB (GPU 0; 11.92 GiB total capacity; 10.92 GiB already allocated; 3.69 MiB free; 11.46 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

hf_t5.D0
^^^^^^^^
    torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 256.00 MiB (GPU 0; 11.92 GiB total capacity; 11.35 GiB already allocated; 218.69 MiB free; 11.35 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

Summary
-------
    Success: 16
    Failures: 4
```

**Review by breuleux — COMMENTED** [review]
ci-cpu and ci-rocm do not appear to be used and I feel they should be left out until they are part of the testing procedure, otherwise we don't really know whether they are good as they are or require adjustments.

The error reporting itself is not tested, but it would be a good idea to do it. I suggest adding a purposely broken configuration file with a single test that has an invalid argument and just adding an additional `milabench run` on that file which only passes if the call has a non-zero error code.

File: tests/config/oom.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    -t: train
+
+
+benchmarks:
+  hf_t5:
+    <<<: *torchbench
+
+    model: hf_T5
+    argv:
+      --bs: 32
```
Comment (breuleux) [inline tests/config/oom.yaml:67]: I think it'd be simpler to write `--bs: "xyz"` to provoke a type error in the argument parser.

File: tests/test_error.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+def test_error_reporting(capsys):
+
+    with pytest.raises(SystemExit) as err:
+        milabench_cmd("run", config("oom"))
+
+    assert err.type is SystemExit
+    assert err.value.code == -1
+
+    captured = capsys.readouterr()
+    assert captured.out == expected_output
```
Comment (breuleux) [inline tests/test_error.py:18]: Add a check on `captured.err` as well, checking that a distinctive part of the expected error output is there is enough.

Comment (Delaunay) [reply tests/test_error.py]: I added a check and stderr is empty in that case

**Review by breuleux — APPROVED** [review]
Looks good to me.

**Pull request merged**